samtranslator/schema/common.py

```diff
@@ -59,3 +59,8 @@ class Config:
 # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html
 class Ref(BaseModel):
     Ref: str
+
+
+class Tags(BaseModel):
```

Comment:
Nice, we can use `Tags` in other schema files.

Reply:
That's the point! I suggested, Prad makes a separate PR for that

Reply:
Where else is this format used? Typically it's a tag key-value map, here it's a list of maps.

Reply:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-graphqlapi.html

It's listed as a list of tag objects here.

Reply:
I mean, why is this in `common`? Which other resources will benefit from this?

Reply:
Both [lambda functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html) and [state machines](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-stepfunctions-statemachine.html) (TagEntrys instead of Tags but they're equivalent shapes) are using lists. I'm not sure why it's split as I can see here that [ApiGateway](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-stepfunctions-statemachine.html ) uses just a dictionary, but I think this will be useful for other resources as we choose to add them or modify typings for existing ones.

Reply:
Those are for CFN resources. The schema here is for SAM resources. If that format is not used commonly for SAM resources, it shouldn't be in `common.py`.

Reply:
Is it not more sensible to have a one to one mapping from SAM to CFN for these cases? It's minor but it sounds like bad UX to me for a customer to change the typing for how they define `Tags` if they are coming from CFN.

Reply:
But also, having a lack of consistency for what is essentially the same property across different resources is also bad UX...

Reply:
Perhaps. My point is mainly just that `Tags` should be in `aws_serverless_graphqlapi.py` unless there are good reasons not to. No SAM resource currently uses lists for tags, so I don't see why this is in `common.py`.

As for which format for tags to use, I'd probably lean toward use whatever CFN uses, as this also gives intrinsics support for keys. But it would set a new precedent, which is probably fine.

Reply:
That is fair. I'll move the `Tags` class to `aws_serverless_graphqlapi.py` for now. We can discuss if we'd like to change our formatting for `Tags` in our SAM resources to match CFN later and we can make any appropriate adjustments based on the outcome.

Reply:
@pradhapanr I think @hoffa raised a good point. List of dictionaries of 2 fields doesn't seem to be an example of great UX. Besides, it will go against UX with other SAM resources. I think it's better to have tags as a simple Dict for SAM resource and then transform it into `[{"key": "xxx", "value": "yyy"},]` for CFN

Reply:
Yeah, so I talked a bit more with @hoffa and it makes sense not to break precedent just for this one resource. There are [reasons](https://github.com/aws/serverless-application-model/issues/2292) to look into using the other way of declaring tags, or possibly supporting both. But that would require more discussion.

I'm going to create another commit with the dictionary typing and adjustments where necessary.